src/cim/components/modals/AddBmcHostYamlForm.tsx

```diff
@@ -0,0 +1,200 @@
+    input.style.visibility = 'hidden';
+    const body = document.querySelector('body');
+    body?.appendChild(input);
+    input.onchange = (e: Event) => {
+      setShowOpeningMessage(true);
+      input.remove();
+      const target = e.target as HTMLInputElement;
+      const file: File = (target.files as FileList)[0];
+      const maxSize = 12582912; //12 Mb
+      if (file.size > maxSize) {
+        setShowOpenFileButton(false);
+        setFileName(file.name);
+        setFileError(t('ai:The file is too big. Upload a file up to 12 Mb.'));
+        setYamlContent(undefined);
+      } else {
+        const reader = new FileReader();
+        reader.readAsText(file, 'UTF-8');
+        reader.onload = (readerEvent) => {
+          const content = readerEvent.target?.result;
+          // parse yaml
+          if (typeof content === 'string') {
+            try {
+              //await new Promise((r) => setTimeout(r, 100));
+              const yamlContent = jsYaml.loadAll(content);
+              setFileName(file.name);
+              setFileError(undefined);
+              setYamlContent(yamlContent);
+              setShowOpenFileButton(false);
+            } catch (err) {
+              setFileName(file.name);
+              setFileError(
+                t(
+                  'ai:File is not structured correctly. Use the template to use the right file structure.',
+                ),
+              );
+              setYamlContent(undefined);
+              setShowOpenFileButton(false);
+            }
+          }
+        };
+      }
+    };
+    input.click();
+  };
+
+  const deleteFileContent = () => {
+    setFileError(undefined);
+    setFileName(undefined);
+    setShowOpenFileButton(true);
+    setShowOpeningMessage(false);
+  };
+
+  const handleSubmit = async () => {
+    try {
+      setShowOpeningMessage(false);
+      setError(undefined);
+      if (yamlContent !== undefined) {
+        await onCreateBmcByYaml(yamlContent);
+      }
+
+      onClose();
+    } catch (e) {
+      setError(getErrorMessage(e));
+    }
+  };
+
+  return (
+    <Form>
+      <>
+        <ModalBoxBody>
+          <Stack hasGutter>
+            <EmptyState variant={EmptyStateVariant.large}>
+              <EmptyStateIcon icon={UploadIcon} />
+              <Title headingLevel="h4" size="lg">
+                {t('ai:Upload a YAML file')}
+              </Title>
+              <EmptyStateBody>
+                <p>
+                  {t(
+                    "ai:Upload a YAML file to add hosts's credentials. For each host, include: hostname, Baseboard Mangement Controller Address, username and password",
+                  )}
+                </p>
+                <StackItem>
+                  <Button variant={ButtonVariant.link} onClick={downloadYaml}>
+                    {t('ai:Download template')}
+                  </Button>
+                </StackItem>
+                {showOpenFileButton && (
+                  <Button
```

Comment:
It may be worth investigating use of PatternFly's file upload components https://www.patternfly.org/v4/components/file-upload#bringing-your-own-file-browse-logic